### PR TITLE
Enable Algolia Docsearch (documentation search)

### LINKED ISSUE #1513: Enable search on the documentation
It is possible to enable search on Docusaurus, using [Algolia DocSearch](https://community.algolia.com/docsearch/): https://docusaurus.io/docs/en/search

CC @kevinbarabash

### PR DESCRIPTION
**Pull request by ylemkimon**
Fixes #1513.

### CODE REVIEW

File: website/siteConfig.js [hunk 1]
```diff
@@ -185,6 +185,14 @@ const siteConfig = {
 
     separateCss: ['static/static', 'static\\static'],
 
+    algolia: {
+        apiKey: '46ecd80046d78d4e5d9a5c06f559dfaa',
```
Comment (kevinbarabash) [inline website/siteConfig.js]: Where did this key come from?  It feels a little unsafe to include a plaintext API key like this.

Comment (ylemkimon) [reply website/siteConfig.js]: @kevinbarabash It is provided Algolia. It is Search-only API Key, which is safe to use in front-end code and in production. Also, Docusaurus documentation suggests entering the API key to its configuration: https://docusaurus.io/docs/en/search.

**Comment by ylemkimon** [comment]
The configuration for our DocSearch is available at: https://github.com/algolia/docsearch-configs/blob/master/configs/katex.json.

One question is should we include supported functions in `table`s, i.e., `td`s?

**Comment by kevinbarabash** [comment]
> One question is should we include supported functions in tables, i.e., tds?

Not sue.  It looks things are set up to use `td` so let's give that a try.  I assume we can change it later if need be.

**Pull request merged**